### PR TITLE
(Proposal) Fix multiple rpc callbacks for same host.

### PR DESCRIPTION
**Pull request by CathalT**
https://github.com/nanocurrency/raiblocks/issues/1073
Instead of looping through all resolved addresses, only iterate if there is a failure and try again.

### CODE REVIEW

**Comment by cryptocode** [comment]
@CathalT any chance you could rebase this?

**Comment by CathalT** [comment]
Will have a look this evening

**Comment by CathalT** [comment]
Apologies lads, bit caught up over Christmas, will get around to it in the next day or two!

File: nano/node/node.cpp [hunk 1]
```diff
@@ -1853,6 +1787,84 @@ nano::node::~node ()
...
+								}
+								else
+								{
+									if (node_l->config.logging.callback_logging ())
+									{
+										BOOST_LOG (node_l->log) << boost::str (boost::format ("Callback to %1%:%2% failed with status: %3%") % address % port % resp->result ());
+									}
+									node_l->stats.inc (nano::stat::type::error, nano::stat::detail::http_callback, nano::stat::dir::out);
+									++i_a;
+									node_l->do_rpc_callback (i_a, address, port, target, body, resolver);
```
Comment (cryptocode) [inline nano/node/node.cpp]: Should this only be done in the `async_connect` callback? Seems like resolving has succeeded if we get to async_write/read.

**Pull request merged**